packages/babel-plugin-material-ui/src/index.ts

```diff
@@ -0,0 +1,134 @@
+import * as babel from '@babel/core';
+import * as t from '@babel/types';
+import resolveImports, { MappedImportsType } from './resolveImports';
+
+let coreImports: MappedImportsType;
+let labImports: MappedImportsType;
+let stylesImports: MappedImportsType;
+
+function handleImports(
+  path: babel.NodePath<t.ImportDeclaration>,
+  importMap: MappedImportsType,
+  rootPath: string,
+) {
+  const { node } = path;
+
+  for (const key in importMap) {
+    if (!importMap.hasOwnProperty(key)) {
+      continue;
+    }
+
+    const elements = importMap[key];
+    const newSpecifiers: Array<t.ImportDefaultSpecifier | t.ImportSpecifier> = [];
+
+    node.specifiers = node.specifiers.filter(spec => {
+      if (!t.isImportSpecifier(spec)) {
+        return true;
+      }
+
+      const mappedImport = elements.find(x => x.name === spec.imported.name);
+      if (!mappedImport) {
+        return true;
+      }
+
+      newSpecifiers.push(
+        mappedImport.default
+          ? t.importDefaultSpecifier(spec.local)
+          : t.importSpecifier(spec.local, spec.imported),
+      );
+      return false;
+    });
+
+    if (newSpecifiers.length) {
+      path.insertBefore(t.importDeclaration(newSpecifiers, t.stringLiteral(`${rootPath}/${key}`)));
+    }
+
+    if (node.specifiers.length === 0) {
+      break;
+    }
+  }
+
+  if (node.specifiers.length === 0) {
+    path.remove();
+  }
+}
+
+function handleIcons(path: babel.NodePath<t.ImportDeclaration>, rootPath: string) {
+  const { node } = path;
+  node.specifiers = node.specifiers.filter(spec => {
+    if (!t.isImportSpecifier(spec)) {
+      return true;
+    }
+
+    path.insertBefore(
+      t.importDeclaration(
+        [t.importDefaultSpecifier(spec.local)],
+        t.stringLiteral(`${rootPath}/${spec.imported.name}`),
+      ),
+    );
+
+    return false;
+  });
+
+  if (node.specifiers.length === 0) {
+    path.remove();
+  }
+}
+
+interface Options {
+  useES: boolean;
```

Comment:
I think that we should remove this option. People that want to use the es module should use an alias in their bundler. I would hope that useES: false + alias can work at the same time. Also, it's probably a recommendation that we should document!

Reply:
Based on https://github.com/mui-org/material-ui/pull/17143#discussion_r317475621 it seems `useESM` can be removed as well.